django_fido/views.py

```diff
@@ -0,0 +1,125 @@
+"""Views for U2F registration and login."""
+from __future__ import unicode_literals
+
+import logging
+from abc import ABCMeta, abstractmethod
+from copy import deepcopy
+
+from django.contrib.auth.mixins import LoginRequiredMixin
+from django.http import JsonResponse
+from django.urls import reverse_lazy
+from django.utils.encoding import force_text
+from django.utils.six import with_metaclass
+from django.utils.translation import ugettext_lazy as _
+from django.views.generic import FormView, View
+from six.moves import http_client
+from six.moves.urllib.parse import urlsplit, urlunsplit
+from u2flib_server import u2f
+
+from .constants import REGISTRATION_REQUEST_SESSION_KEY, U2F_REGISTRATION_REQUEST
+from .forms import U2fResponseForm
+from .models import U2fDevice
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class BaseU2fRequestView(with_metaclass(ABCMeta, View)):
+    """Base view for U2F request views.
+
+    @cvar session_key: Session key where the U2F request is stored.
+    @cvar u2f_request_factory: Function which accepts `app_id` and `registered_keys` and returns U2F request.
+    """
+
+    session_key = None
+    u2f_request_factory = None
+
+    def get_app_id(self):
+        """Return appId - base URL to the web."""
+        # Correct `appId` for the web application doesn't have a path.
+        # See https://developers.yubico.com/U2F/App_ID.html
+        chunks = urlsplit(self.request.build_absolute_uri('/'))
+        return urlunsplit((chunks.scheme, chunks.netloc, '', '', ''))
+
+    @abstractmethod
+    def get_user(self):
+        """Return user which is subject of the request."""
+        raise NotImplementedError
```

Comment:
Having `pass` here might be safer. If the concrete class calls `super()` (which is possible...), this would get thrown.

Reply:
OK, that looks like a common pattern :) Fixed.